.github/workflows/publish.yml

```diff
@@ -27,7 +31,13 @@ jobs:
       - name: Set up Gradle
         uses: gradle/gradle-build-action@v2
       - name: Use tag as version
+        # Set a production version only when running on tags
+        if: ${{ github.event_name != 'pull_request' }}
         run: echo "${GITHUB_REF#refs/*/}" > version.txt
+      - name: Use dummy version for PR
+        # Set a dummy version that doesn't hurt if accidentally released
+        if: ${{ github.event_name == 'pull_request' }}
+        run: echo "0.0.1-SNAPSHOT" > version.txt
```

Comment:
I think it would be nicer to have only one step with an `if` inside of the `run`, so we don't have at least one skipped step for every execution.